python/pyspark/sql/tests/test_dataframe.py

```diff
@@ -759,6 +759,30 @@ def test_create_dataframe_from_pandas_with_dst(self):
                 os.environ['TZ'] = orig_env_tz
             time.tzset()
 
+    # SPARK-35211: inferred schema verification
+    @unittest.skipIf(not have_pandas, pandas_requirement_message)  # type: ignore
+    def test_create_dataframe_from_pandas_with_mismatched_udt(self):
+        from pyspark.testing.sqlutils import ExamplePoint
+        import pandas as pd
+        pdf = pd.DataFrame({'point': pd.Series([ExamplePoint(1, 1), ExamplePoint(2, 2)])})
+        # The underlying sqlType of ExamplePoint is ArrayType(DoubleType(), False)
+        # There is a data type mismatch between 1 and DoubleType, a TypeError is expected
+        with self.assertRaises(TypeError):
+            with self.sql_conf({"spark.sql.execution.arrow.pyspark.enabled": "false"}):
+                self.spark.createDataFrame(pdf)
+        # With verifySchema disabled, there will be unexpected behaviours
+        self.assertEquals(self.spark.createDataFrame(pdf, verifySchema=False).collect(),
+                          [Row(point=ExamplePoint(0.0, 0.0)), Row(point=ExamplePoint(0.0, 0.0))])
```

Comment:
Can you add a positive test case where `verifySchema` is `True` and results are correct?

Reply:
added